### PR TITLE
Build distroless kubectl image

### PR DESCRIPTION
**Pull request by holgero**
The current "distroless" kubectl image (used to install crds in helm) is not really distroless, but based on ubuntu.

And to help us figure out who should review this PR, please put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[x] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

**Comment by holgero** [comment]
Hmm, the distroless test shows a few problems:
* A container spec that uses the kubectl image to run a 'run.sh' script:
```yaml
      serviceAccountName: istio-security-post-install-account
      containers:
        - name: kubectl
          image: "istio-testing/kubectl:istio-testing-distroless"
          imagePullPolicy: IfNotPresent
          command: [ "/bin/bash", "/tmp/security/run.sh", "/tmp/security/custom-resources.yaml" ]
          volumeMounts:
            - mountPath: "/tmp/security"
              name: tmp-configmap-security
```
and here the `run.sh` from a config map:
```bash
    #!/bin/sh
    
    set -x
    
    if [ "$#" -ne "1" ]; then
        echo "first argument should be path to custom resource yaml"
        exit 1
    fi
    
    pathToResourceYAML=${1}
    
    kubectl get validatingwebhookconfiguration istio-galley 2>/dev/null
    if [ "$?" -eq 0 ]; then
        echo "istio-galley validatingwebhookconfiguration found - waiting for istio-galley deployment to be ready"
        while true; do
            kubectl -n istio-system get deployment istio-galley 2>/dev/null
            if [ "$?" -eq 0 ]; then
                break
            fi
            sleep 1
        done
        kubectl -n istio-system rollout status deployment istio-galley
        if [ "$?" -ne 0 ]; then
            echo "istio-galley deployment rollout status check failed"
            exit 1
        fi
        echo "istio-galley deployment ready for configuration validation"
    fi
    sleep 5
    kubectl apply -f ${pathToResourceYAML}
```
* Another container spec that puts a script into the command for the kubectl image:
```yaml
      serviceAccountName: istio-cleanup-secrets-service-account
      containers:
        - name: kubectl
          image: "istio-testing/kubectl:istio-testing-distroless"
          imagePullPolicy: IfNotPresent
          command:
          - /bin/bash
          - -c
          - >
              kubectl get secret --all-namespaces | grep "istio.io/key-and-cert" |  while read -r entry; do
                ns=$(echo $entry | awk '{print $1}');
                name=$(echo $entry | awk '{print $2}');
                kubectl delete secret $name -n $ns;
              done
```

Seems making this image distroless will be a bit more difficult.

File: docker/Dockerfile.kubectl [hunk 1]
```diff
@@ -1,11 +1,25 @@
+# BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
+ARG BASE_DISTRIBUTION=default
+
 # hadolint ignore=DL3006
-FROM istionightly/base_debug
-# Image for post install jobs
+FROM istionightly/base_debug as default
 
 # This container should only contain kubectl.  Hard-coding to use Linux K8s 1.11.1 version.
 ADD https://storage.googleapis.com/kubernetes-release/release/v1.11.1/bin/linux/amd64/kubectl /usr/bin/kubectl
```
Comment (howardjohn) [inline docker/Dockerfile.kubectl]: as a followup we may want to bump the version, we only support k8s 1.12-1.15 now

**Comment by howardjohn** [comment]
Didn't look at tests, distroless is failing with https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio/16372/e2e-simpleTests-distroless-master/1492/artifacts/simple-auth-test-faf7bd0e91984aa780a3/istio-security-post-install-istio-testing-distroless-qtqml_describe.log



We should probably make this a required test, I think today it is an agenda item in T&R meeting, I will propose this

**Comment by howardjohn** [comment]
distroless test is now required!

**Comment by howardjohn** [comment]
Seems like we need to revert https://github.com/istio/istio/pull/12173/ to do this

**Comment by holgero** [comment]
It's a bit more complicated to get rid of the shell snippets and run the stuff really distroless. Closing this for now.

**Pull request closed without merging**